Makefile

```diff
@@ -16,14 +16,25 @@ ZMQ_VERSION ?= 4.3.1
 ZMQ_BASE_BUILD_DIR = $(CURDIR)/libzmq/build
 ZMQ_BUILD_DIR = $(ZMQ_BASE_BUILD_DIR)/v$(ZMQ_VERSION)
 ZMQ_PKG_CONFIG_DIR = $(ZMQ_BUILD_DIR)/lib/pkgconfig
+ifeq ($(OS),Windows_NT)
```

Comment:
I think we should separate the setting of `CXXFLAGS` and `SHARED_EXT` and only set `ZMQ_COMPILE_HOST` here.

And then have a separate conditional afterwards which checks the value of `ZMQ_COMPILE_HOST` to set `CXXFLAGS` and `SHARED_EXT`. This way when setting `ZMQ_COMPILE_HOST` in Travis we can still cross compile since on Travis `$(OS)` won't be `Windows_NT`. You can use `$(findstring "mingw", $(ZMQ_COMPILE_HOST))` for the test.

Also see the comment below on `ZMQ_COMPILE_HOST`.